### ISSUE TITLE
Customization Failing on OOBE Template

### ISSUE THREAD

**Issue by luckerby** [terraform-providers/terraform-provider-vsphere#695]
I'm using a simple config file to deploy just 1 VM using terraform + the Vsphere provider. This worked just fine until now, by cloning a Windows Server 2016 box and doing some basic post-config (name the machine in Windows, assign IP/DNS list, etc).

I've recently added a new step to the procedure I'm following, whereby the initial machine that's cloned has been converted to a template, and then this was [sysprepped](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/sysprep--system-preparation--overview), and configured to enter OOBE (out-of-box experience). The trouble is that right after terraform instantiates the new VM, it complains according to the 'Actual behavior' section below, indicating that the Network side of the guest OS is at fault. I just can't figure out the issue.

### Terraform Version
Terraform v0.11.11

### vSphere Provider Version
1.9.0

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files

Relevant configuration follows

```hcl
data "vsphere_network" "network" {
  name          = "Servers"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
...
resource {
    network_interface {
      network_id = "${data.vsphere_network.network.id}"
    }
    clone {
      template_uuid = "${data.vsphere_virtual_machine.template.id}"
      
      customize {
          network_interface {
              ipv4_address = "192.168.0.100"
              ipv4_netmask = 24
          }
          ipv4_gateway = "192.168.01"
      }
}
```

### Actual Behavior

```
vsphere_virtual_machine.vm: Still creating... (40s elapsed)
vsphere_virtual_machine.vm: Still creating... (50s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.vm: 1 error(s) occurred:

* vsphere_virtual_machine.vm:
Virtual machine customization failed on "/xxxx/xxxxx/vm/GVA-terraform-test1":

An error occurred while setting up network properties of the guest OS. See the log file C:/Windows/TEMP/vmware-imc/guestcust.log in the guest OS for details.

The virtual machine has not been deleted to assist with troubleshooting. If
corrective steps are taken without modifying the "customize" block of the
resource configuration, the resource will need to be tainted before trying
again. For more information on how to do this, see the following page:
https://www.terraform.io/docs/commands/taint.html


Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

The new VM is left in this state:
![image](https://user-images.githubusercontent.com/31319583/51999846-5cc33e00-24c4-11e9-8244-a7fa3593cb4c.png)

The content of the file referenced in the `terraform apply` output is [here](https://gist.github.com/luckerby/92bcbc6cb9aa72066fc2c19b4ff1fbdf).

**Comment by luckerby** [comment]
Skipping the sysprep results in the VMs being deployed just as expected using terraform. However there's a slight difference if the sysprepped template is deployed manually, as to processed and deployed by terrafrom, as follows:

C:\Windows\Temp on the manually deployed sysprepped machine:
![image](https://user-images.githubusercontent.com/31319583/52478042-c4f0de80-2bac-11e9-809b-cc8d94d2c7b8.png)

C:\Windows\Temp on the terraform-deployed sysprepped machine:
![image](https://user-images.githubusercontent.com/31319583/52478081-e6ea6100-2bac-11e9-85f4-534681d3c9a4.png)

The vmware-imc folder containing the file that lists the detailed errors is only to be found on the VM deployed by terraform, not on the manually deployed one. I'm trying to figure out where the issue lies exactly - either with terraform + Vmware, or with Vmware itself. At this point, I'm afraid that simply opening a ticket with Vmware will be ignored, based on the fact that the manually deployed one doesn't throw any error.

**Comment by bill-rich** [comment]
Hi again @luckerby! When you deploy manually through vCenter, the same sysprepped template that is failing with through Terraform is working? If so, would it be possible to get the customization spec exported and posted to see if something is being done differently in the VMware generated spec? 

Another interesting test could be to try cloning the VM through `govc` and specifying the customization spec that was generated in vCenter.

**Comment by rismoney** [comment]
You don't need to run sysprep.  Guest customization takes care of this, and will work if you have vmtools loaded in the image.

I wget an unattended file in setupcomplete, that will be processed for oobe.  Your regional settings probably need to be in there as the system resets those.

**Comment by luckerby** [comment]
Just tested by deploying 2 VMs using terraform using a template as source (that wasn't sysprepped - just as @rismoney mentioned), and indeed the SIDs are different:
![image](https://user-images.githubusercontent.com/31319583/52577328-85cfc100-2e2a-11e9-8791-8aae917fa9d7.png)

I will do some more tests and post back tomorrow.

**Comment by luckerby** [comment]
I've tested further, and regardless of using a regular - as in just installed and simply shut down - either simple VM or template, the machine SID will be regenerated for every new machine that terraform deploys.

It appears this was discovered before, albeit by trial and error. Someone [here](https://www.virtualizationhowto.com/2018/06/sysprep-and-vmware-guest-customization-with-terraform/) did the same thing as me and actually sysprepped the source, then ran into issues. He specifically mentions that as long as `skip_customization = true` is not specified in the config file, the machine SID will be regenerated. There's also a past issue about the same topic [here](https://github.com/terraform-providers/terraform-provider-vsphere/issues/336).

Perhaps it will be better to explicitly state the fact that the machine SID is regenerated as part of the [provider's documentation](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html).

Bottom line - I wrongly assumed Sysprep has to be specifically performed, since terraform wouldn't do it (or the actions behind it) on its own. This turned out to be false.

@rismoney - thank you for making me challenge my initial wrong assumption
@bill-rich - thank you for reopening the issue. I'm leaving this one opened right now in case it's used as a reminder to update the docs, should this be the case.

**Comment by bill-rich** [comment]
Thanks @luckerby! I've got a PR open to document this.
